pkg/liquidity-mining/contracts/gauges/RewardsOnlyGauge.vy

```diff
@@ -0,0 +1,613 @@
+# @version 0.3.1
+"""
+@title Rewards-Only Gauge
+@author Curve Finance
+@license MIT
+@notice Distribution of third-party rewards without CRV
+"""
+
+from vyper.interfaces import ERC20
+
+implements: ERC20
+
+struct RewardToken:
+    distributor: address
+    period_finish: uint256
+    rate: uint256
+    duration: uint256
+    received: uint256
+    paid: uint256
+
+interface Streamer:
+    def reward_data(_token: address) -> RewardToken: view
+    def last_update_time() -> uint256: view
+
+interface ERC20Extended:
+    def symbol() -> String[26]: view
+
+interface ERC1271:
+    def isValidSignature(_hash: bytes32, _signature: Bytes[65]) -> bytes32: view
+
+event Deposit:
+    provider: indexed(address)
+    value: uint256
+
+event Withdraw:
+    provider: indexed(address)
+    value: uint256
+
+event Transfer:
+    _from: indexed(address)
+    _to: indexed(address)
+    _value: uint256
+
+event Approval:
+    _owner: indexed(address)
+    _spender: indexed(address)
+    _value: uint256
+
+struct Reward:
```

Comment:
Is this just to return a bunch of data in one go?

Reply:
Yes, this is so that the `reward_data(address)` function behaves the same on `RewardsOnlyGauge` and `LiquidityGaugeV5` to make things easier on the frontend otherwise they'd have to handle external rewards differently on mainnet vs polygon.